core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -380,6 +381,23 @@ public void removeBlocks(List<Long> blockIds, boolean delete) throws Unavailable
     }
   }
 
+  @Override
+  public void validateBlocks(Function<Long, Boolean> validator, boolean repair)
+      throws UnavailableException {
+    List<Long> invalidBlocks = new ArrayList<>();
+    for (long blockId : mBlocks.keySet()) {
+      if (!validator.apply(blockId)) {
+        LOG.warn("Block {} has no corresponding file metadata. Restart Alluxio with the startup "
```

Comment:
It is strange to me that the block master is logging something about files. Shouldn't this type of log message be in the validator, since the validator knows about files (and the block master does not)?

Reply:
Good points, updated thanks.